### ISSUE TITLE
update rule: react-a11y-anchors - checks should span files

### ISSUE THREAD

**Issue by HamletDRC** [microsoft/tslint-microsoft-contrib#232]
These two checks should keep state across multiple files: 

Links with the same HREF should have the same link text. 
- (http://oaa-accessibility.org/wcag20/rule/39/)

Links that point to different HREFs should have different link text. 
- (http://oaa-accessibility.org/wcag20/rule/41/)

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
